deepface/models/demography/Age.py

```diff
@@ -37,11 +41,64 @@ def __init__(self):
         self.model = load_model()
         self.model_name = "Age"
 
-    def predict(self, img: np.ndarray) -> np.float64:
-        # model.predict causes memory issue when it is called in a for loop
-        # age_predictions = self.model.predict(img, verbose=0)[0, :]
-        age_predictions = self.model(img, training=False).numpy()[0, :]
-        return find_apparent_age(age_predictions)
+    def predict(self, img: Union[np.ndarray, List[np.ndarray]]) -> np.ndarray:
+        """
+        Predict apparent age(s) for single or multiple faces
+        Args:
+            img: Single image as np.ndarray (224, 224, 3) or
+                List of images as List[np.ndarray] or
+                Batch of images as np.ndarray (n, 224, 224, 3)
+        Returns:
+            np.ndarray (n,)
+        """
+        # Convert to numpy array if input is list
+        if isinstance(img, list):
+            imgs = np.array(img)
+        else:
+            imgs = img
+
+        # Remove batch dimension if exists
+        imgs = imgs.squeeze()
```

Comment:
you squeezed imgs and then expand its dimension, instead can we decide the input is single or many images when predict called?

```python
if isinstance(img, list):
   # then it is many images
elif len(img.shape) == 3:
   # then it is single image
elif len(img.shape) == 4 and img.shape[0] == 1:
   # then it is single image
elif len(img.shape) == 4 and img.shape[0] > 1:
   # then it is many images
else:
   raise error


```